src/proteus/dswx_hls.py

```diff
@@ -4222,11 +4255,52 @@ class in the browse image. False to display PSW-Agg.
     logger.info(f'    mean azimuth angle: {sun_azimuth_angle}')
     logger.info(f'    mean elevation angle: {sun_elevation_angle}')
 
-    logger.info(f'Data coverage:')
-    logger.info(f'    Input HLS product spatial coverage [%]:'
-                f' {hls_product_spatial_coverage}')
-    logger.info(f'    Input HLS product cloud coverage [%]:'
-                f' {hls_product_cloud_coverage}')
+    # print input HLS product spatial and cloud coverage
+    logger.info(f'data coverage:')
+    if 'INPUT_HLS_PRODUCT_SPATIAL_COVERAGE' in dswx_metadata_dict.keys():
+        hls_product_spatial_coverage = dswx_metadata_dict[
+            'INPUT_HLS_PRODUCT_SPATIAL_COVERAGE']
+        logger.info(f'    input HLS product spatial coverage [%]:'
+                    f' {hls_product_spatial_coverage}')
+
+    if 'INPUT_HLS_PRODUCT_CLOUD_COVERAGE' in dswx_metadata_dict.keys():
+        hls_product_cloud_coverage = dswx_metadata_dict[
+            'INPUT_HLS_PRODUCT_CLOUD_COVERAGE']
+        logger.info(f'    input HLS product cloud coverage [%]:'
+                    f' {hls_product_cloud_coverage}')
+
+    # compute a preliminary cloud layer, i.e., without snow/ice class
+    preliminary_cloud_layer = _compute_preliminary_cloud_layer(
+        fmask, mask_adjacent_to_cloud_mode)
```

Comment:
I've gone back and forth about this preliminary cloud layer. It was really confusing to me until I spent a bit of time with the code, and I really do not like splitting the logic for the CLOUD layer into two separate places; the docstrings are confusing/redundant, and creates extra steps that need to be tracked and followed up on later in the code.

What about a new approach: 

1) Put all logic for creating the CLOUD layer (cloud+shadow+adj+snow) into a new function, e.g. `_compute_cloud_layer()`.
2) Call that function at this point in the code.
3) When computing the statistic of `CLOUD_COVERAGE`, it would become something like:
```
n_cloud_and_valid = np.sum(((preliminary_cloud_layer) != 0) & 
                                                    (preliminary_cloud_layer) != SNOW) & 
                                                    (valid_array))
```
4) Later, turn the function `_compute_and_apply_cloud_layer()` into simply `_compute_wtr_2_layer(wtr_1_layer=..., cloud_layer=...)`

This approach would greatly simplify, streamline, and modularize the code, reduce docstring redundancies and errors, and help future maintainers see exactly where/how each layer is being computed.

Reply:
Thank you, @nemo794 . The` mask_adjacent_to_cloud_mode`  `cover` requires the WTR-2 layer to update the CLOUD layer with snow/ice values. I update the docstrings and answered some of your comments above. If it's still unclear, we can meet tomorrow (Tuesday Jan 31st) to solve this issue.